src/services/fs/dir_stream.rs

```diff
@@ -77,16 +87,16 @@ impl ObjectPage for DirPager {
 }
 
 pub struct BlockingDirPager {
-    root: String,
+    root: PathBuf,
 
     size: usize,
     rd: std::fs::ReadDir,
 }
 
 impl BlockingDirPager {
-    pub fn new(root: &str, rd: std::fs::ReadDir) -> Self {
+    pub fn new(root: impl Into<PathBuf>, rd: std::fs::ReadDir) -> Self {
```

Comment:
I prefer to use `&PathBuf` or `PathBuf`.

Reply:
Done in ee705357979c02ee6d83913f24d567df80a065a4